node/src/main/scala/com/wavesplatform/api/http/eth/EthRpcRoute.scala

```diff
@@ -69,12 +68,14 @@ class EthRpcRoute(blockchain: Blockchain, transactionsApi: CommonTransactionsApi
         case "eth_getTransactionCount" =>
           resp(id, quantity(time.getTimestamp()))
         case "eth_getBlockByNumber" =>
-          resp(
-            id,
-            Json.obj(
-              "number" -> quantity(Integer.parseInt(param1Str.drop(2), 16))
-            )
-          )
+          val blockNumber = param1Str match {
+            case "earliest" => Some(1)
+            case "latest"   => Some(blockchain.height)
+            case "pending"  => None
```

Comment:
According to the task 
```suggestion
            case "pending"  => Some(blockchain.height)
```